### PR TITLE
Multi-Tenant SSO

### PR DESCRIPTION
**Pull request by swainn**
Adds support for a mutli-tenant SSO authentication flow for four providers: AD FS, Azure AD, OneLogin, and Okta.

multi-tenant-tenant-page

* Adds a new custom view to Tethys that displays a form asking user to enter tenant. Includes a feature to save the tenant for the future using localStorage in the browser.
* Implements custom auth and complete views that check for multi-tenant conditions and add tenant to the backend for use looking up tenant-specific settings later on.
* Completes implementation of tenant view so that it calls do_auth when tenant info has been set on the backend.
* Adds SOCIAL_AUTH_FIELDS_STORED_IN_SESSION to settings.py so that the tenant field of the tenant form will be saved in the session storage. The value is retrieved in complete view to load the tenant settings on callback.
* Adds MultiTenantMixin class that is used to override the backend ``settings`` method.
* Adds documentation for multi-tenant configuration to SSO docs.
* Adds new settings to the Tethys Portal Configuration docs.
* Adds tests for all new functionality

### CODE REVIEW

File: tethys_portal/static/tethys_portal/js/sso_tenant.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    // Check local storage for saved data
+    let backend = $('#backend').val();
+    let remember_key = `${backend}-remember`;
+    let tenant_key = `${backend}-tenant`;
+    let remember = localStorage.getItem(remember_key);
+    let tenant = localStorage.getItem(tenant_key);
+    console.log(backend);
+    console.log(remember_key);
+    console.log(tenant_key);
```
Comment (ezrajrice) [inline tethys_portal/static/tethys_portal/js/sso_tenant.js:14]: Are you planning on keeping these console.logs in?

Comment (ezrajrice) [inline tethys_portal/static/tethys_portal/js/sso_tenant.js:14]: Is there a reason to leave these logs in the live version?

**Review by ezrajrice — CHANGES_REQUESTED** [review]

File: tethys_portal/static/tethys_portal/js/sso_tenant.js (hunk 1 above)
Comment (swainn) [reply tethys_portal/static/tethys_portal/js/sso_tenant.js]: Nope. Just removed them.

**Comment by swainn** [comment]
@ezrajrice, btw this PR is still WIP b/c I'm awaiting Gage's review of a PR that adds the documentation. I'll remove WIP once it is ready to go.

**Comment by swainn** [comment]
@ezrajrice this is ready to go. Take a look at the documentation that I added and let me know if any parts need clarification.

**Review by ezrajrice — APPROVED** [review]
Approved.

**Comment by swainn** [comment]
@ezrajrice I just pushed a new commit to address the issue with username length discrepancies between the User model and the form field. I also updated the max_length on all email, first_name, and last_name fields while I was at it. Take a look.

**Review by sdc50 — APPROVED** [review]

**Review by ezrajrice — APPROVED** [review]

**Pull request merged**